pkg/proxy/util/utils.go

```diff
@@ -78,6 +78,38 @@ func BuildPortsToEndpointsMap(endpoints *v1.Endpoints) map[string][]string {
 	return portsToEndpoints
 }
 
+// ContainsIPv4Loopback returns true if the input is empty or one of the CIDR contains an IPv4 loopback address.
+func ContainsIPv4Loopback(cidrStrings []string) bool {
+	if len(cidrStrings) == 0 {
+		return true
+	}
+	// RFC 5735 127.0.0.0/8 - This block is assigned for use as the Internet host loopback address
+	ipv4LoopbackStart := netutils.ParseIPSloppy("127.0.0.0")
+	ipv4LoopbackEnd := netutils.ParseIPSloppy("127.255.255.255")
+	for _, cidr := range cidrStrings {
+		if IsZeroCIDR(cidr) {
+			return true
+		}
+
+		ip, ipnet, err := netutils.ParseCIDRSloppy(cidr)
+		if err != nil {
+			continue
+		}
+
+		if netutils.IsIPv6CIDR(ipnet) {
+			continue
+		}
+
+		if ip.IsLoopback() {
+			return true
+		}
+		if ipnet.Contains(ipv4LoopbackStart) || ipnet.Contains(ipv4LoopbackEnd) {
```

Comment:
Do you need both?  How could it contain ipv4LoopbackEnd but not ipv4LoopbackStart and not be `IsLoopback()` ?

Reply:
honestly, I wasn't doing the bit operations but it can't, right?
to "contain" the other subnet, the mask has to be always greater and will completely contain the network ... I don't know why I had a memory flash that this can happen ... let me fix it